pkg/manager/internal.go

```diff
@@ -87,9 +92,16 @@ type controllerManager struct {
 	// metricsListener is used to serve prometheus metrics
 	metricsListener net.Listener
 
+	// metricsInsecureServing enables insecure metrics serving.
+	// This means metrics will be served via http and without authentication and authorization.
+	metricsInsecureServing bool
+
 	// metricsExtraHandlers contains extra handlers to register on http server that serves metrics.
 	metricsExtraHandlers map[string]http.Handler
 
+	// metricsKubeClient is the client used to authenticate and authorize requests to the metrics endpoint.
+	metricsKubeClient *kubernetes.Clientset
```

Comment:
Using the clientset like this rather than clients for the specific APIs means that we will end up importing all clients for all apis, which we should try to avoid